src/raftstore/store/store.rs

```diff
@@ -483,26 +488,57 @@ impl<T: Transport, C: PdClient> Store<T, C> {
         let ids: Vec<u64> = self.pending_raft_groups.drain().collect();
         let pending_count = ids.len();
 
-        for region_id in ids {
-            let mut ready_result = None;
-            if let Some(peer) = self.region_peers.get_mut(&region_id) {
-                match peer.handle_raft_ready(&self.trans) {
-                    Err(e) => {
-                        // TODO: should we panic or shutdown the store?
-                        error!("handle raft ready at region {} err: {:?}", region_id, e);
-                        return Err(e);
+        let mut sent_cnt = 0;
+        let (tx, rx) = mpsc::channel();
+        for &region_id in &ids {
+            if let Some(mut peer) = self.region_peers.remove(&region_id) {
+                let sender = tx.clone();
+                let trans = self.trans.clone();
+                self.apply_pool.execute(move || {
```

Comment:
apply seems only to apply raft logs, but here we need to handle other raft steps too.